lib/src/touch_spin.dart

```diff
@@ -68,40 +68,38 @@ class _TouchSpinState extends State<TouchSpin> {
           padding: widget.iconPadding,
           iconSize: widget.iconSize,
           color: minusBtnDisabled
-              ? widget.iconDisabledColor ?? Theme.of(context).disabledColor
-              : widget.iconActiveColor ??
-                  Theme.of(context).textTheme.button.color,
+              ? widget.iconDisabledColor
+              : widget.iconActiveColor,
           icon: widget.subtractIcon,
           onPressed: minusBtnDisabled
               ? null
               : () {
                   num newVal = _value - widget.step;
-                  setState(() {
-                    _value = newVal;
-                  });
-                  if (widget.onChanged != null) widget.onChanged(newVal);
+
+                  setState(() => _value = newVal);
+
+                  widget.onChanged(newVal);
```

Comment:
The null guard was important because `onChanged` is _optional_.

A concise alternative is:
```dart
widget.onChanged?.call(newVal);
```